### ISSUE TITLE
The relationship between event.metadata['agent']['rotation'], event.metadata['agent']['cameraHorizon'] and camera orientation?

### ISSUE THREAD

**Issue by HuynhLam** [allenai/ai2thor#152]
Hi,

1. What is the relationship between (`event.metadata['agent']['rotation']` + `event.metadata['agent']['cameraHorizon']`) and the camera orientation?
If I use e.g. Euler angle as my orientational representation, how do these metadata values are related to (pitch, yaw, roll)?

2. How does the camera pose change If I perform `controller.step(dict(action='TeleportFull', x=point['x'], y=point['y'], z=point['z'], rotation=rot, horizon=gor))` ?

3. If I do `controller.step(dict(action='AddThirdPartyCamera', rotation=dict(x=hor, y=rot, z=0), position=dict(x=point['x'], z=point['z'], y=point['y'])))`, what will I get with 
`event.metadata['thirdPartyCameras'][frame_number]['position']` and
`event.metadata['thirdPartyCameras'][frame_number]['rotation']`?

4. Do values from question 3 similar to `event.metadata['agent']['rotation']` and `event.metadata['agent']['cameraHorizon']`?

Thank you very much.

**Comment by ekolve** [comment]
1) The agent's rotation is a value from 0 - 360 that represents the agent's rotation about the Y axis in global space.  It is independent from the camerHorizon.  The cameraHorizon represents the rotation of the camera about the local X axis of the agent.  Take a look at the following lines of code to see how both of these are updated:
https://github.com/allenai/ai2thor/blob/master/unity/Assets/Scripts/BaseFPSAgentController.cs#L860

2) The agent will be moved to the X,Y,Z position in global space and the rotation about the Y axis set to the value 'rot', the camera's rotation about the local X axis will set to 'gor'. Take a look at this line:
https://github.com/allenai/ai2thor/blob/master/unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs#L1056.  That line should explain what you are after (it only applies when you forceAction=True, but the logic is the same if you don't have an item in your hand).

3) event.metadata['thirdPartyCameras'] is a list dicts that contain the position, rotation and cameraId of each camera that has been added. I just added some documentation here: http://ai2thor.allenai.org/documentation/event-metadata.  The position/rotation corresponds to the position and rotation of the camera in global space.

4) The value event.metadata['thirdPartyCameras'][0]['rotation']['x'] would correspond to the value for the agent's cameraHorizon value.
